src/mapping/AxialGeoMultiscaleMapping.cpp

```diff
@@ -0,0 +1,173 @@
+#include "AxialGeoMultiscaleMapping.hpp"
+#include <Eigen/src/Core/Matrix.h>
+#include "mesh/Mesh.hpp"
+
+namespace precice::mapping {
+
+AxialGeoMultiscaleMapping::AxialGeoMultiscaleMapping(
+    Constraint     constraint,
+    int            dimensions,
+    MultiscaleType type,
+    MultiscaleAxis axis,
+    double         radius)
+    : Mapping(constraint, dimensions, false, Mapping::InitialGuessRequirement::None),
+      _type(type),
+      _axis(axis),
+      _radius(radius)
+{
+  setInputRequirement(Mapping::MeshRequirement::VERTEX);
+  setOutputRequirement(Mapping::MeshRequirement::VERTEX);
+}
+
+void AxialGeoMultiscaleMapping::computeMapping()
+{
+  PRECICE_TRACE(output()->vertices().size());
+
+  PRECICE_ASSERT(input().get() != nullptr);
+  PRECICE_ASSERT(output().get() != nullptr);
+
+  if (getConstraint() == CONSISTENT) {
+    PRECICE_DEBUG("Compute consistent mapping");
+    if (_type == SPREAD) {
+      PRECICE_CHECK(input()->vertices().size() == 1, "You can only define an axial geometric multiscale mapping of type spread from a mesh with exactly one vertex.");
+
+      /* When we add support for 1D meshes (https://github.com/precice/precice/issues/1669),
+        we should check for valid dimensions combination.
+
+        const int inValueDimensions  = input()->getDimensions();
+        const int outValueDimensions = output()->getDimensions();
+
+        PRECICE_CHECK(input()->getDimensions() == 1, "The input mesh on an axial geometric multiscale mapping can only be 1D at the moment, but it was defined to be {}.", input()->getDimensions())
+        PRECICE_CHECK(output()->getDimensions() == 3, "The output mesh on an axial geometric multiscale mapping can only be 3D at the moment, but it was defined to be {}.", input()->getDimensions())
+      */
+      const int outValueDimensions = 3;
+
+      int effectiveCoordinate = _axis;
+      PRECICE_ASSERT(effectiveCoordinate == MultiscaleAxis::X ||
+                         effectiveCoordinate == MultiscaleAxis::Y ||
+                         effectiveCoordinate == MultiscaleAxis::Z,
+                     "Unknown multiscale axis type.")
+
+      // compute distances between 1D vertex and 3D vertices
+      mesh::Vertex &v0      = input()->vertices()[0];
+      size_t const  outSize = output()->vertices().size();
+
+      for (size_t i = 0; i < outSize; i++) {
```

Comment:
I didn't know of the `Index::getVerticesInsideBox(const mesh::Vertex &centerVertex, double radius)` method. This would indeed make collecting vertices easier. However:

- We do use these distances later in the mapping
- We would need to provide an alternative implementation of `getVerticesInsideBox`, which includes the boundary. Currently:

```c++
  /// Return all the vertices inside the box formed by vertex and radius (boundary exclusive)
  std::vector<VertexID> getVerticesInsideBox(const mesh::Vertex &centerVertex, double radius);
```

Reply:
Is this necessary though?

You use $\frac{distance}{radius} \le 1.05$ or $distance \le 1.05 \times radius$ as a cut-off.

So, you could query `getVerticesInsideBox(v, 1.05*radius+math::NUMERICA_ZERO_DIFFERENCE)` and filter them, or implement a `getVerticesInsideSphere(v, 1.05*radius+math)`.